File: Time_Needed_to_Buy_Tickets.go
package main

import (
	"fmt"
)

func main() {
	tickets := []int{5, 1, 1, 1}
	k := 0
	fmt.Println(timeRequiredToBuy(tickets, k))
}

func timeRequiredToBuy(tickets []int, k int) int {
	t := 0
	for tickets[k] != 0 {
		for i := 0; i < len(tickets); i++ {
			if tickets[i] != 0 && tickets[k] != 0 {
				tickets[i] -= 1
				t += 1
			}
		}
	}

	return t
}

File: Time_Needed_to_Buy_Tickets/Time_Needed_to_Buy_Tickets.py

tickets = [5,1,1,1]
k = 0
def timeRequiredToBuy(ickets, k):
    t=0
    while tickets[k] != 0:
        for i in range(len(tickets)):
            if tickets[i] !=0 and tickets[k] !=0:
                tickets[i] -= 1
                t+=1
    return t

print(timeRequiredToBuy(tickets,k))